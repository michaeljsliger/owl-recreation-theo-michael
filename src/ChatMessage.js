import React from 'react';
// import ChatLog from './ChatLog';
import STORE from './STORE';
// import chatEvents from './ChatSTORE';



const findParticipantNameByChatId = (id) => STORE.find(el => el.id === parseInt(id))


class ChatMessage extends React.Component {
    
    // what props does this take in?
    // id, message, time, type
    render() {
        const currName = findParticipantNameByChatId(this.props['data-id']);
        

        const Emoted = function(type) {
            switch(type) {
                case 'thumbs-up':
                    return 'gave a thumbs up!';
                case 'thumbs-down':
                    return 'gave a thumbs down!';
                case 'raise-hand':
                    return 'raised their hand!';
                case 'clap':
                    return 'clapped';
                case 'join':
                    return 'joined';
                case 'leave':
                    return 'left';
                case 'join-stage':
                    return 'joined the stage';
                case 'leave-stage':
                    return 'leaved the stage';
                default:
                    return 'Emoted!';
            }
        }

        var myDate = new Date(this.props.time);
        const timing = myDate.toLocaleString();

        if (this.props.type === 'message') {
             return (
                <div className="single-message">
                    <div className="single-message-header">
                        <div><h5>{currName.name}</h5></div>
                        <div><h6>{timing}</h6></div>
                    </div>
                    <div className="time-container">
                        <p>{this.props.message}</p>
                    </div>
                </div>
             )} else {
                 return (
                   <div className="single-message">
                    {currName.name} {Emoted(this.props.type)}
                   </div>  
                 )
             }
    }
}

export default ChatMessage;