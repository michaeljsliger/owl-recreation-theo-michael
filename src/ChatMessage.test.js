import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import ChatMessage from './ChatMessage';
import renderer from 'react-test-renderer';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

describe('Chat Message Branch Testing', () => {
  // smoke test

  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatMessage data-id="3" />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  // snapshot test
  it('should render correctly', () => {
    const tree = renderer
          .create(<ChatMessage type="message" data-id="3" />)
          .toJSON();
    expect(tree).toMatchSnapshot();
  })
})