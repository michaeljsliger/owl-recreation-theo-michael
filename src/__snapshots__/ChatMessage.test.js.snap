// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Message Branch Testing should render correctly 1`] = `
<div
  className="single-message"
>
  <div
    className="single-message-header"
  >
    <div>
      <h5>
        Axe Kubicka
      </h5>
    </div>
    <div>
      <h6>
        Invalid Date
      </h6>
    </div>
  </div>
  <div
    className="time-container"
  >
    <p />
  </div>
</div>
`;
